### PR TITLE
Add git clean script that preserves auth files

### PR DESCRIPTION
**Pull request by juliettepretot**
# Checklist

- [ ] Pull request affects core Oak functionality (e.g. runtime, SDK, ABI)
  - [ ] I have written tests that cover the code changes.
  - [ ] I have checked that these tests are run by
        [Cloudbuild](/cloudbuild.yaml)
  - [ ] I have updated [documentation](/docs/) accordingly.
  - [ ] I have raised an [issue](https://github.com/project-oak/oak/issues) to
        cover any TODOs and/or unfinished work.
- [ ] Pull request includes prototype/experimental work that is under
      construction.

### CODE REVIEW

File: scripts/git_clean [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+readonly SCRIPTS_DIR="$(dirname "$0")"
+# shellcheck source=scripts/common
+source "$SCRIPTS_DIR/common"
+
+# Stash untracked but not ignored files
+git stash push --include-untracked
+
+# Delete remaining untracked files (defacto only ignored files), excluding auth
+# files. Use double --force to delete untracked nested repositories.
+git clean -d --force --force -x --exclude="/.oak_remote_cache_key.json" --exclude="**/client_secret_*.apps.googleusercontent.com.json"
```
Comment (ipetr0v) [inline scripts/git_clean:14]: Why not use `-X` for only ignored files?

Comment (juliettepretot) [reply scripts/git_clean:14]: Using that option overwrites the `--exclude` option.

Comment (ipetr0v) [inline scripts/git_clean:14]: Are these `.json` files everything we want to keep?

**Review by ipetr0v — APPROVED** [review]

File: scripts/git_clean (hunk 1 above)
Comment (juliettepretot) [reply scripts/git_clean:14]: Seems so to me, but if there's other stuff people want to keep I'm happy to add'em.

**Review by conradgrobler — APPROVED** [review]

**Pull request merged**